# HG changeset patch
# User Betterbird <betterbird@betterbird.eu>
# Date 1753537025 -7200
# Parent  a1ecb3fa81b317e6420f4e36b5c545d79a63a16f
Feature: Open Activity Manager in tab, introduce pref mail.activity.managerInTab.

diff --git a/mail/app/profile/all-thunderbird.js b/mail/app/profile/all-thunderbird.js
--- a/mail/app/profile/all-thunderbird.js
+++ b/mail/app/profile/all-thunderbird.js
@@ -1377,8 +1377,9 @@ pref("pdfjs.firstRun", true);
 pref("pdfjs.previousHandler.preferredAction", 0);
 pref("pdfjs.previousHandler.alwaysAskBeforeHandling", false);
 
 pref("mail.activity.loglevel", "Warn");
+pref("mail.activity.managerInTab", true);
 
 // The number of public recipients before we offer BCC addressing.
 pref("mail.compose.warn_public_recipients.threshold", 15);
 
diff --git a/mail/base/content/mailCore.js b/mail/base/content/mailCore.js
--- a/mail/base/content/mailCore.js
+++ b/mail/base/content/mailCore.js
@@ -613,9 +613,15 @@ function openAddonsMgr(aView) {
   });
 }
 
 function openActivityMgr() {
-  openContentTab("chrome://messenger/content/activity.xhtml");
+  if (Services.prefs.getBoolPref("mail.activity.managerInTab", true)) {
+    openContentTab("chrome://messenger/content/activity.xhtml");
+  } else {
+    Cc["@mozilla.org/activity-manager-ui;1"]
+      .getService(Ci.nsIActivityManagerUI)
+      .show(window);
+  }
 }
 
 /**
  * Open the folder properties of current folder with the quota tab selected.
